### PR TITLE
Rework handling of images and versions

### PR DESCRIPTION
**Pull request by dtantsur**
User-supplied images via CRD's goes against operator best practices and
make upgrades very complicated. Remove the Images structure but allow
for downstream customizations via CLI flags and environment variables.

Record the version that was installed to use it in upgrade logic.

Signed-off-by: Dmitry Tantsur <dtantsur@protonmail.com>

### CODE REVIEW

File: pkg/ironic/version.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package ironic
+
+const (
+	installedVersion string = "latest"
```
Comment (elfosardo) [inline pkg/ironic/version.go:4]: wondering if we want to have a different name for this (e.g. ironicInstalledVersion) as we talked about releasing ipa-downloader, and in the future we may need another const for that

**Review by elfosardo — COMMENTED** [review]
looks good
just a comment before adding LGTM

File: pkg/ironic/version.go (hunk 1 above)
Comment (dtantsur) [reply pkg/ironic/version.go:4]: Possibly, but it's a private variable, we can deal with it later.

**Pull request merged**